runtime/shibuya/src/lib.rs

```diff
@@ -1535,6 +1539,70 @@ impl pallet_migrations::Config for Runtime {
     type WeightInfo = pallet_migrations::weights::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+    pub const EnterDuration: BlockNumber = 4 * HOURS;
+    pub const EnterDepositAmount: Option<Balance> = None;
+    pub const ExtendDuration: BlockNumber = 2 * HOURS;
+    pub const ExtendDepositAmount: Option<Balance> = None;
+    pub const ReleaseDelay: Option<u32> = None;
+}
+
+/// Calls that can bypass the safe-mode pallet.
+pub struct SafeModeWhitelistedCalls;
+impl Contains<RuntimeCall> for SafeModeWhitelistedCalls {
+    fn contains(call: &RuntimeCall) -> bool {
+        match call {
+            RuntimeCall::Sudo(_)
+            // System and Timestamp are required for block production
+            | RuntimeCall::System(_)
+            | RuntimeCall::Timestamp(_)
+            | RuntimeCall::SafeMode(_)
+            | RuntimeCall::TxPause(_) => true,
+            _ => false,
+        }
+    }
+}
+
+/// Calls that cannot be paused by the tx-pause pallet.
+pub struct TxPauseWhitelistedCalls;
+/// All calls can be paused, except Sudo calls.
+impl Contains<RuntimeCallNameOf<Runtime>> for TxPauseWhitelistedCalls {
+    fn contains(full_name: &pallet_tx_pause::RuntimeCallNameOf<Runtime>) -> bool {
+        matches!(full_name.0.as_slice(), b"Sudo")
+    }
+}
+
+impl pallet_safe_mode::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type Currency = Balances;
+    type RuntimeHoldReason = RuntimeHoldReason;
+    type WhitelistedCalls = SafeModeWhitelistedCalls;
+    type EnterDuration = EnterDuration;
+    type EnterDepositAmount = EnterDepositAmount;
+    type ExtendDuration = ExtendDuration;
+    type ExtendDepositAmount = ExtendDepositAmount;
+    // The 'Success' value below represents the number of blocks that the origin may induce safe mode
+    type ForceEnterOrigin =
+        EnsureWithSuccess<EnsureRootOrAllTechnicalCommittee, AccountId, EnterDuration>;
+    type ForceExtendOrigin =
+        EnsureWithSuccess<EnsureRootOrAllTechnicalCommittee, AccountId, ExtendDuration>;
+    type ForceExitOrigin = EnsureRootOrAllTechnicalCommittee;
+    type ForceDepositOrigin = EnsureRootOrAllTechnicalCommittee;
```

Comment:
Up for discussion, but I'd lean for 2/3 here.

Reply:
For the exit and release of deposits, we can stick to a quorum of 2/3, or even a unanimous quorum, to ensure that things are properly corrected. I have chosen 2/3 now in this PR.

Reply:
For Astar definitely, I merely suggested 1/2 because it's what we're using anyways already on Shibuya. Not important though since we probably won't test it there.